commerce-payment-method-mercanet/src/main/java/com/liferay/commerce/payment/method/mercanet/internal/servlet/MercanetServlet.java

```diff
@@ -131,17 +135,28 @@ protected void doPost(
 				String redirect = ParamUtil.getString(
 					httpServletRequest, "redirect");
 
+				if (!Objects.equals(
+						parameterMap.get("responseCode"), "00")) {
+
+					String orderId = parameterMap.get("orderId");
+
+					Long commerceOrderId = Long.parseLong(orderId);
```

Comment:
This should be `long commerceOrderId` -- as it is, `Long.parseLong` returns a `long` which gets autoboxed to a `Long` because of the type of the variable, and then below gets autounboxed to a `long` again when passed to `cancelPayment`. Plus Liferay style just never uses `Long`s unless absolutely necessary :)

Generally it would also be better to use `GetterUtil.getLong` rather than `Long.parseLong`, but it depends on what we would want to happen if `orderId` is not numeric in some edge case. `Long.parseLong` would throw a `NumberFormatException`, while `GetterUtil.getLong` would return the default long (0 or whatever you pass in). Not sure what the best behavior would be here.